### PR TITLE
Add default label to Fleet Secrets to enable BRO backups

### PR DESCRIPTION
**Pull request by mallardduck**
### Summary
This PR fixes SURE-6919
Fixes https://github.com/rancher/dashboard/issues/11211

There were recent fixes to the BRO codebase that corrected a bug which backed up far more data than it should have been backing up. In other terms, the resourceset filters BRO uses to collect resources for backups are properly followed now.

As such, the only fleet secrets that get backed up would be ones with the `"fleet.cattle.io/managed=true"` label. Since that is the existing rule defined for fleet within the resoruceset.

### Technical notes summary

This PR aims to make sure that the secrets being created by the fleet UI are labeled as managed by fleet. This in turn will allow BROs rules to collect the secrets for backup.

This change is not retroactive - any existing secrets should be updated by Rancher admins to include the correct label.

Additionally, we are intentionally not addressing when users select a secret created outside of the UI for fleet. This maybe worth looking into expanding on later, but for now the BRO team perspective is that those would be "user-created" (and managed) resources; not something Rancher created for them.

### Areas or cases that should be tested

1. Create a GitRepo and define a new secret via the GitRepo UI,
2. Find the secrets name after creation,
3. Explore secrets in the local cluster to find new secret,
4. Observe it currently lacks the expected label.

After this PR it should be observed that the secret will have the expected label.
Once complete this will allow testing of the main issue per direction on https://github.com/rancher/dashboard/issues/11211

### Areas which could experience regressions

Unlikely to occur.

### Screenshot/Video

N/A. There are no visual UX changes.

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or **there are no UX changes**

### CODE REVIEW

File: shell/edit/fleet.cattle.io.gitrepo.vue [hunk 1]
```diff
@@ -413,8 +413,9 @@ export default {
           type:     SECRET,
           metadata: {
             namespace:    this.value.metadata.namespace,
-            generateName: 'auth-'
-          },
+            generateName: 'auth-',
+            labels:       { 'fleet.cattle.io/managed': 'true' }
```
Comment (richard-cox) [inline shell/edit/fleet.cattle.io.gitrepo.vue:417]: This should reference https://github.com/rancher/dashboard/blob/53544f5d190ba53dd7b6c819208732efe30c2ee1/shell/config/labels-annotations.js#L113

**Review by richard-cox — CHANGES_REQUESTED** [review]
@mallardduck Thanks for creating this PR. Can you make sure you run through the PR checklist?

**Comment by nwmac** [comment]
This also needs automated tests to checks that the annotation is applied in the API request that the UI sends.

File: shell/edit/fleet.cattle.io.gitrepo.vue (hunk 1 above)
Comment (mallardduck) [reply shell/edit/fleet.cattle.io.gitrepo.vue]: Appreciate it - fixed this now.

File: shell/edit/fleet.cattle.io.gitrepo.vue [hunk 2]
```diff
@@ -22,7 +22,7 @@ import { base64Decode, base64Encode } from '@shell/utils/crypto';
 import SelectOrCreateAuthSecret from '@shell/components/form/SelectOrCreateAuthSecret';
 import { _CREATE } from '@shell/config/query-params';
 import { isHarvesterCluster } from '@shell/utils/cluster';
-import { CAPI, CATALOG } from '@shell/config/labels-annotations';
+import { CAPI, CATALOG, FLEET as FLEET_LABELS } from '@shell/config/labels-annotations';
```
Comment (mallardduck) [inline shell/edit/fleet.cattle.io.gitrepo.vue:25]: Used `as` since there's already a `FLEET` import.

**Comment by mallardduck** [comment]
So pending successful run of `e2e-test (admin, @adminUser, @fleet)` which will have the tests I modified to verify the new label - I think this is good for review now. I adjusted the test to verify that the request being sent and the secret in the response both have the label we expect. Assuming I read the chai docs right then I expect the new assertion on the tests will be valid now, so I'll check that box as well.

**Comment by mallardduck** [comment]
Added @joshmeranda as a Reviewer for O&B team perspective.

**Review by joshmeranda — APPROVED** [review]
LGTM, obviously would like to get eyes from UI

**Review by richard-cox — APPROVED** [review]
I've created https://github.com/rancher/dashboard/issues/11222 for some follow on UX changes. Currently we hide all system based labels and annotations, and in some cases it would be good to see them

**Pull request merged**